drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -636,29 +639,149 @@ static int bq274xx_gauge_configure(const struct device *dev)
 		return -EIO;
 	}
 
+#ifdef CONFIG_PM_DEVICE
+	bq274xx->pm_state = DEVICE_PM_ACTIVE_STATE;
+#endif
+
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int bq274xx_enter_shutdown_mode(struct bq274xx_data *data)
+{
+	int status = 0;
+
+	status = bq274xx_control_reg_write(data, BQ274XX_UNSEAL_KEY);
+	if (status < 0) {
+		LOG_ERR("Unable to unseal the battery");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data, BQ274XX_UNSEAL_KEY);
+	if (status < 0) {
+		LOG_ERR("Unable to unseal the battery");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data,
+					   BQ274XX_CONTROL_SHUTDOWN_ENABLE);
+	if (status < 0) {
+		LOG_ERR("Unable to enable shutdown mode");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data, BQ274XX_CONTROL_SHUTDOWN);
+	if (status < 0) {
+		LOG_ERR("Unable to enter shutdown mode");
+		return -EIO;
+	}
+
+	status = bq274xx_control_reg_write(data, BQ274XX_CONTROL_SEALED);
+	if (status < 0) {
+		LOG_ERR("Failed to seal the gauge");
+		return -EIO;
+	}
+
+	data->pm_state = DEVICE_PM_OFF_STATE;
+
+	return 0;
+}
+
+static int bq274xx_exit_shutdown_mode(const struct device *dev)
+{
+	struct bq274xx_data *bq274xx = dev->data;
+	const struct bq274xx_config *const config = dev->config;
+
+	bq274xx->gpio = device_get_binding((char *)config->gpio_port);
+	if (!bq274xx->gpio) {
+		LOG_DBG("Gpio controller %s not found.", config->gpio_port);
+		return -EINVAL;
+	}
+
+	gpio_pin_configure(bq274xx->gpio, config->int_pin,
+			   GPIO_OUTPUT | GPIO_OPEN_DRAIN);
+
+	gpio_pin_set(bq274xx->gpio, config->int_pin, 1);
+	k_usleep(200);
+	gpio_pin_set(bq274xx->gpio, config->int_pin, 0);
+
+	gpio_pin_configure(bq274xx->gpio, config->int_pin,
+			   GPIO_INPUT | GPIO_OPEN_DRAIN | config->int_flags);
+
+	int status = bq274xx_gauge_init(dev);
+
```

Comment:
remove empty line

Reply:
The empty line wasn't there in the first place, but I had to put it, otherwise the compliance check fails here.

Reply:
then compliance need's to be fixed, not new line.

Reply:
Just declare `int status;` at the top of the block.

Reply:
Moved to the top.